TIGLCreator/src/TIGLCreatorSelectWingAndFlapStatusDialog.cpp

```diff
@@ -124,27 +124,59 @@ void TIGLCreatorSelectWingAndFlapStatusDialog::cleanup()
     return;
 }
 
-double TIGLCreatorSelectWingAndFlapStatusDialog::getTrailingEdgeFlapValue( std::string uid )
+double TIGLCreatorSelectWingAndFlapStatusDialog::getTrailingEdgeFlapValue(std::string uid)
+{
+    try {
+        auto it = _deviceMap.find(uid);
+        if (it == _deviceMap.end()) {
+            throw tigl::CTiglError("getTrailingEdgeFlapValue: UID not found", TIGL_UID_ERROR);
+        }
+
+        tigl::CTiglAbstractGeometricComponent* controlSurfaceDevice = it->second;
+        tigl::CCPACSTrailingEdgeDevice* device = dynamic_cast<tigl::CCPACSTrailingEdgeDevice*>(controlSurfaceDevice);
+
+        return device->GetControlParameter();
+    }
+    catch (...) {
+        return 0;
```

Comment:
This try-catch block that swallows all exceptions and pretends like nothing happened looks a bit sketchy to me. I know, not your code, but let's use the opportunity to clean up. To the very least, we should catch CTiglError and throw an exception, if the try-catch block is needed here at all.

The same holds for the getLeadingEdgeFlapValue